packages/@react-aria/dnd/src/useDrag.ts

```diff
@@ -150,14 +154,25 @@ export function useDrag(options: DragOptions): DragResult {
   };
 
   let onDragEnd = (e: DragEvent) => {
-    if (typeof options.onDragEnd === 'function') {
-      options.onDragEnd({
-        type: 'dragend',
-        x: e.clientX,
-        y: e.clientY,
-        dropOperation: DROP_EFFECT_TO_DROP_OPERATION[e.dataTransfer.dropEffect]
-      });
-    }
+    let event: DragEndEvent = {
+      type: 'dragend',
+      x: e.clientX,
+      y: e.clientY,
+      dropOperation: DROP_EFFECT_TO_DROP_OPERATION[e.dataTransfer.dropEffect]
+    };
+
+    setTimeout(() => {
```

Comment:
See if I'll still need this when I move global state to aria

Reply:
With the subsequent changes made when moving the global state to aria, we technically don't need the timeout except if we are worried about the following case:
1. user drags and drops a item from a draggable collection to a droppable collection
2. onDragEnd fires first since it doesn't have a timeout that delays it. The item from the draggable collection is removed in onDragEnd/onRemove
3. onDrop occurs after the set timeout. It attempts to look up the item from the draggable collection in order to add it to the droppable collection but can't find it since it was deleted in step 2.

This may or may not be a typical case since the end user may instead just perform a "move" operation in their database or use the information passed in via items/datatransfer to construct the dragged item instead

Reply:
We've made changes to get rid of the setTimeouts in useDrag and useDrop, opting to fire onDragEnd if the drag element was removed by onDrop